File: python/section4/q32.py
from q30 import load_neko
import pandas as pd
from pprint import pprint

neko = load_neko()
varbs = set()
for sentence in neko:
    morphos = pd.DataFrame(sentence)
    varbs.update(
        morphos[morphos['pos'] == '動詞']['base'].tolist()
    )
pprint(varbs)

File: scala/section4/src/main/scala/q32.scala
/**
  * Created by nomotoeriko on 2017/04/24.
  */
object q32 {
  def main(args: Array[String]): Unit = {
    val neko = q30.load_neko("../../data/neko.txt.mecab")
    val varbs = get_varb(neko)
    varbs.foreach(println)
  }

  def get_varb(neko: List[List[Map[String, String]]]): Set[String] = {
    neko.map(sentence => sentence.filter(morph => morph("pos")=="動詞").map(_("base"))).reduce(_:::_).toSet
  }
}
